### DISCUSSION TITLE
Auto-fill checkered rectangle area with pure white color after DEL rectangle area?

### DISCUSSION THREAD

**Discussion by pstein** [tannerhelland/PhotoDemon#528] [General]
Assume I load a PNG image, then select a rectangle area and press DEL key.
The rectangle area is checkered now.
Now I select another rectangle area or another tool.

I expected that the previous rectangle area is automatically adjusted to the default pure white background.
But it isn't.

Unfortunately sub menus like
- Layer-->Merge Down
or
- Image--->Flatten Image 
are greyed out/disabled.

How can I (at least) manually or automatically (preferred) switch checkered area to pure white?

**Comment by tannerhelland** [comment]
Hello @pstein .  It sounds like you are new to working with layers in an image editor.

When you erase a region of an image, that region is now transparent (empty).  All image editors use a checkerboard to convey that a region is "empty".  Think of it as cutting a hole in the image.

To fill that region, you have many options.

- You can use the `Layer > Transparency > Remove Transparency` command to remove transparency without saving the file.
![image](https://github.com/tannerhelland/PhotoDemon/assets/1930029/9817e7c6-0b7a-4da4-bf80-0840efc37e4b)

- You can place a new background layer (in whatever color you want) via `Layer > Add > Basic Layer` .  Just make sure the layer is *below* the current layer and it will fill in any transparent regions.
![image](https://github.com/tannerhelland/PhotoDemon/assets/1930029/da35fcde-34d1-4099-a061-085c9817094b)

- If you save the image to a format that does *not* support transparency (like JPG), PhotoDemon will automatically provide an option for background color.
![image](https://github.com/tannerhelland/PhotoDemon/assets/1930029/b4887f90-790e-40d1-8d11-811695029dd5)

- If you save the image to a format that *does* support transparency (like PNG), but you don't *want* transparency, you can use the advanced save options to turn off transparency and replace it with a background color.
![image](https://github.com/tannerhelland/PhotoDemon/assets/1930029/e5fb8222-de9a-4d4d-b5dc-fb09bea9ff49)

Hope that helps!

**Reply by pstein** [reply]
Thank you.

So it seems there is no option to let PD AUTOMATICALLY remove transparency when I select something else?
Can this be added (at least as an option)?

**Comment by hi5** [comment]
Perhaps as a suggestion, keep DEL as it currently is as it is established behaviour, but say when pressing <kbd>ctrl</kbd>+<kbd>del</kbd> the area is filled with the foreground colour?

**Reply by pstein** [reply]
Seems not to help.
After pressing CTRL+DEL the background is still checkered.

How do I find out the current foreground and background color?

**Reply by hi5** [reply]
That was my suggestion for a new feature, so it isn't possible yet, but should the developer consider it, using a modifier key + del might be a way to implement it without the need for "settings" or additional buttons/dialogs.

As far as I know PD just uses foreground color and has no foreground/background color as most other editors do. (Something I do miss at times) as it is tempting to paint with the right mouse button to paint with the "background color" for quick(er) colour fills)
